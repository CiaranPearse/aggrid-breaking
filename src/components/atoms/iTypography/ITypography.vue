<template>
  <div :class="classes">{{ content }}</div>
</template>

<script lang="ts">
import { computed } from 'vue';
import { Appearances, Alignments, Emphasis } from '../../../constants';

export default {
  name: 'ITypography',
  props: {
    content: {
      type: String,
      required: true,
    },
    appearance: {
      type: String,
      default: Appearances.Body1,
    },
    align: {
      type: String,
      default: Alignments.Left,
    },
    emphasis: {
      type: String,
      default: Emphasis.Regular,
    }
  },

  setup(props: { content: string; appearance: string; align: string; emphasis:string;}) {
    return {
      classes: computed(() => ({
        'typography-text': true,
        'text-h1': props.appearance === Appearances.H1,
        'text-h2': props.appearance === Appearances.H2,
        'text-h3': props.appearance === Appearances.H3,
        'text-h4': props.appearance === Appearances.H4,
        'text-h5': props.appearance === Appearances.H5,
        'text-h6': props.appearance === Appearances.H6,
        [`text-${props.appearance || 'text-body-1'}`]: true,
        'text-left': props.align === 'left',
        'text-center': props.align === 'center',
        'text-right': props.align === 'right',
        'text-justify': props.align === 'justify',
        'text-start': props.align === 'start',
        'text-end': props.align === 'end',
        'font-weight-black': props.emphasis === Emphasis.Bolder,
        'font-weight-bold': props.emphasis === Emphasis.Bold,
        'font-weight-medium': props.emphasis === Emphasis.Medium,
        'font-weight-regular': props.emphasis === Emphasis.Regular,
        'font-weight-light': props.emphasis === Emphasis.Light,
        'font-weight-thin': props.emphasis === Emphasis.Thin,
        'font-italic': props.emphasis === Emphasis.Italic,
      })),
    };
  },
};
</script>
