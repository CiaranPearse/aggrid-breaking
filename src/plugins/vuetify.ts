import '@mdi/font/css/materialdesignicons.css';
import 'vuetify/styles';
import { createVuetify } from 'vuetify';
import * as components from 'vuetify/components';
import * as directives from 'vuetify/directives';

export const defaultLight = {
  dark: false,
  colors: {
    background: 'light-blue',
    layoutBackground: '#0071c5',
    surface: '#FFFFFF',
    primary: '#0071c5',
    'primary-darken-1': '#3700B3',
    secondary: '#03DAC6',
    'secondary-darken-1': '#018786',
    error: '#B00020',
    info: '#2196F3',
    success: '#4CAF50',
    warning: '#FB8C00',
  },
};
export const darkOne = {
  dark: true,
  colors: {
    background: '#FFFFFF',
    layoutBackground: '#98c3e3',
    surface: '#FFFFFF',
    primary: '#0071c5',
    'primary-darken-1': '#3700B3',
    secondary: '#03DAC6',
    'secondary-darken-1': '#018786',
    error: '#B00020',
    info: '#2196F3',
    success: '#4CAF50',
    warning: '#FB8C00',
  },
};

export default createVuetify({
  components,
  directives,
  theme: {
    defaultTheme: 'defaultLight',
    themes: {
      defaultLight,
      darkOne,
    },
  },
});
