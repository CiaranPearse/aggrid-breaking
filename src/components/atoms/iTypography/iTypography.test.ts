import { createVuetify } from "vuetify";
import { mount } from '@vue/test-utils';
import ITypography from './ITypography.vue';

const vuetify = createVuetify();

describe('Typography', () => {
  
  it('should render', () => {
    const wrapper = mount(ITypography, {
      props: {
          content: "Test Button",
      },
      global: {
          plugins: [vuetify],
      },
  });
    expect(ITypography).toBeTruthy()
  })

  it('renders with Click Me', () => {
    const buttonText = 'Click Me';
    const wrapper = mount(ITypography, {
      props: {
        content: buttonText
      },
      global: {
        plugins: [vuetify],
      },
    });
    expect(wrapper.text()).toMatch(buttonText);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders with Click Me as Caption', () => {
    const buttonText = 'Click Me';
    const wrapper = mount(ITypography, {
      props: {
        content: buttonText,
        appearance: 'caption'
      },
      global: {
        plugins: [vuetify],
      },
    });
    const byClass = wrapper.find('.text-caption')
    expect(wrapper.text()).toMatch(buttonText);
    const correctClass = wrapper.find('.text-caption')
    expect(correctClass.exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot();
  });


  it('renders with Click Me as body-1', () => {
    const buttonText = 'Click Me';
    const wrapper = mount(ITypography, {
      props: {
        content: buttonText,
        align: 'center',
      },
      global: {
        plugins: [vuetify],
      },
    });
    expect(wrapper.text()).toMatch(buttonText);
    const textClass = wrapper.find('.text-body-1');
    const alignClass = wrapper.find('.text-center');
    expect(textClass.exists()).toBe(true)
    expect(alignClass.exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders with Click Me as body-1 and bolder', () => {
    const buttonText = 'Click Me';
    const wrapper = mount(ITypography, {
      props: {
        content: buttonText,
        align: 'center',
        emphasis: 'bolder',
      },
      global: {
        plugins: [vuetify],
      },
    });
    expect(wrapper.text()).toMatch(buttonText);
    const textClass = wrapper.find('.text-body-1');
    const alignClass = wrapper.find('.text-center');
    const emphasisClass = wrapper.find('.font-weight-black');
    expect(textClass.exists()).toBe(true);
    expect(alignClass.exists()).toBe(true);
    expect(emphasisClass.exists()).toBe(false);
    expect(wrapper.html()).toMatchSnapshot();
  });
})