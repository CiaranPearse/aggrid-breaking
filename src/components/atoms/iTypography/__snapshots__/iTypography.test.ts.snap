// Vitest Snapshot v1

exports[`Typography > renders with Click Me 1`] = `"<div class=\\"typography-text text-body-1 text-left font-weight-regular\\">Click Me</div>"`;

exports[`Typography > renders with Click Me as Caption 1`] = `"<div class=\\"typography-text text-caption text-left font-weight-regular\\">Click Me</div>"`;

exports[`Typography > renders with Click Me as body-1 1`] = `"<div class=\\"typography-text text-body-1 text-center font-weight-regular\\">Click Me</div>"`;

exports[`Typography > renders with Click Me as body-1 and bolder 1`] = `"<div class=\\"typography-text text-body-1 text-center\\">Click Me</div>"`;
